src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureVMSqlServerKeyVaultCredentialConfig.cs

```diff
@@ -0,0 +1,140 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using System.Security;
+using System.Security.Permissions;
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Azure.Commands.Compute
+{
+    /// <summary>
+    /// Helper cmdlet to construct instance of SQL Credential AKV based settings class
+    /// </summary>
+    [Cmdlet(
+         VerbsCommon.New,
+         AzureVMSqlServerKeyVaultCredentialConfigNoun),
+     OutputType(
+         typeof(KeyVaultCredentialSettings))]
+    public class NewAzureVMSqlServerKeyVaultCredentialConfigCommand : PSCmdlet
+    {
+        /// <summary>
+        /// Configuration object friendly name
+        /// </summary>
+        protected const string AzureVMSqlServerKeyVaultCredentialConfigNoun = "AzureVMSqlServerKeyVaultCredentialConfig";
+
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 1,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Enable Key Vault Credential.")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter Enable { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 2,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "SQL Server credential name to create.")]
+        [ValidateNotNullOrEmpty]
+        public string CredentialName { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 3,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Azure Key Vault service URL")]
+        [ValidateNotNullOrEmpty]
+        public string AzureKeyVaultUrl { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 4,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Principal user client identifier.")]
+        [ValidateNotNullOrEmpty]
+        public string ServicePrincipalName { get; set; }
```

Comment:
This is already implemented and working for SQL in classic. This not a net new code. We are just enabling for ARM as it already working for Classic. We will consider it on the next set of cmdlet as we are planning a new set of powershel cmdlets